zap/src/main/java/org/parosproxy/paros/network/HttpBody.java

```diff
@@ -557,7 +558,8 @@ public void setCharset(String charsetName) {
             setCharsetImpl(null);
             return;
         }
-
+        charsetName = StringUtils.remove(charsetName, ",text/html");
+        charsetName.toUpperCase();
```

Comment:
toUpperCase() just returns upper case, it doesnt change the string its invoked on. 
Best to just add this to the line above?

Reply:
IMHO I don't think this should be added, the existing code copes with wrong charsets and we shouldn't be adding these one offs to workaround broken servers.

Reply:
If this is actually causing problems better provide more information about the actual issue.

Reply:
The wrong charset got through "utf-8,text/html"  was trying to be passed to that. Which was causing an exception and was potentially causing the code to stop processing while loading a saved session.

Reply:
Please provide a thread dump when that hang happens.

Reply:
Which exception was thrown?

Reply:
That means it's one of the exceptions being caught. In which case did the entire loading fail?

Reply:
At a certain point it was causing the entire session to fail to load. and it was handing in the open method during the InvokeAndWait. 
The InvokeLater as well as the additional code changes addressed all of this. 
If you want I can test it again to see if it loads now without this.

Reply:
I'd suggest trying without the changes in `HttpBody` and `Session`, if it still hangs obtain a thread dump which should give more details about the issue.

Reply:
Is there an easy way to do this ? I will most likely kill this pull request and start fresh from the upstream repo, if you are ok with that.

Reply:
Yea, sorry about that. Been awhile since I used github and or Java..

Reply:
No worries.

Reply:
Getting a unit test failure now, `[org.parosproxy.paros.core.scanner.UtilUnitTest](file:///zap/zaproxy/zap/build/reports/tests/test/classes/org.parosproxy.paros.core.scanner.UtilUnitTest.html)`
However, I did not change anything in that code. Did something change recently?

Reply:
No, that's a timing issue, bound to happen sometimes.